File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/list_meetings.py
#!/usr/bin/env python3
"""
List all upcoming meetings for the authenticated user.
Usage: python list_meetings.py
"""

import json
import requests
import os

# Load config
with open('config.json', 'r') as f:
    config = json.load(f)

ACCOUNT_ID = config['account_id']
CLIENT_ID = config['client_id']
CLIENT_SECRET = config['client_secret']
USER_ID = config['user_id']

def get_access_token():
    url = "https://api.zoom.us/oauth/token"
    headers = {
        "Content-Type": "application/x-www-form-urlencoded"
    }
    data = {
        "grant_type": "account_credentials",
        "account_id": ACCOUNT_ID
    }
    response = requests.post(url, headers=headers, data=data, auth=(CLIENT_ID, CLIENT_SECRET))
    response.raise_for_status()
    return response.json()['access_token']

def list_meetings(access_token):
    url = f"https://api-us.zoom.us/v2/users/{USER_ID}/meetings"
    headers = {
        "Authorization": f"Bearer {access_token}"
    }
    params = {
        "type": "upcoming"
    }
    response = requests.get(url, headers=headers, params=params)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    try:
        token = get_access_token()
        meetings = list_meetings(token)
        
        if 'meetings' not in meetings or not meetings['meetings']:
            print("No upcoming meetings found.")
        else:
            print(f"Found {len(meetings['meetings'])} upcoming meeting(s):")
            for meeting in meetings['meetings']:
                print(f"- ID: {meeting['id']}")
                print(f"  Topic: {meeting['topic']}")
                print(f"  Start: {meeting['start_time']}")
                print(f"  Join URL: {meeting['join_url']}\n")
                
    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")

File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/list_meetings.js
#!/usr/bin/env node
const fs = require('fs').promises;
const path = require('path');
const fetch = require('node-fetch');

// Load config
async function loadConfig() {
  const configPath = path.join(__dirname, '..', 'config.json');
  const data = await fs.readFile(configPath, 'utf8');
  return JSON.parse(data);
}

async function getAccessToken(config) {
  const url = 'https://api.zoom.us/oauth/token';
  const params = new URLSearchParams({
    grant_type: 'account_credentials',
    account_id: config.account_id
  });

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Authorization': `Basic ${Buffer.from(`${config.client_id}:${config.client_secret}`).toString('base64')}`
    },
    body: params.toString()
  });

  if (!response.ok) {
    throw new Error(`Failed to get access token: ${response.status} ${response.statusText}`);
  }

  const data = await response.json();
  return data.access_token;
}

async function listMeetings(config, accessToken) {
  const url = `https://api-us.zoom.us/v2/users/${config.user_id}/meetings?type=upcoming`;
  const response = await fetch(url, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${accessToken}`,
      'Content-Type': 'application/json'
    }
  });

  if (!response.ok) {
    throw new Error(`Failed to list meetings: ${response.status} ${response.statusText}`);
  }

  return await response.json();
}

async function main() {
  try {
    const config = await loadConfig();
    const token = await getAccessToken(config);
    const meetings = await listMeetings(config, token);

    if (!meetings.meetings || meetings.meetings.length === 0) {
      console.log('No upcoming meetings found.');
      return;
    }

    console.log(`Found ${meetings.meetings.length} upcoming meeting(s):`);
    for (const meeting of meetings.meetings) {
      console.log(`- ID: ${meeting.id}`);
      console.log(`  Topic: ${meeting.topic}`);
      console.log(`  Start: ${meeting.start_time}`);
      console.log(`  Join URL: ${meeting.join_url}\n`);
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();